File: CivitasJava/src/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla{

    CasillaImpuesto(float cantidad, String nombre){
        super(nombre);
        importe=cantidad;
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            todos.get(actual).puedeComprar=false;
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }

    @Override
    public String toString(){

        String str="******************************";
        str+="\n " + getNombre() + "    |    " + importe;
        str+="\n******************************\n";
    
        return str;
    }
}

File: CivitasRuby/lib/casilla_impuesto.rb
#encoding:utf-8

require_relative 'casilla'
require_relative 'jugador'
require 'casilla.rb'

module Civitas
    class CasillaImpuesto < Casilla
        def initialize(cantidad, nombre)
            super(nombre)
            @importe=cantidad
        end

        def recibe_jugador(actual, todos)
            if (jugador_correcto(actual,todos))
                todos[actual].puede_comprar=false
                informe(actual,todos)
                todos[actual].paga_impuesto(@importe)
            end
        end

        def to_string  
          String::new(str="**************************************************")
          str+="\n " + @nombre + "    |    " + @importe
          str+="\n**************************************************\n"
          
          return str
        end
    end
end